aten/src/ATen/native/mkldnn/xpu/detail/WoQMatmul.cpp

```diff
@@ -0,0 +1,306 @@
+#include <c10/xpu/XPUFunctions.h>
+
+#include <ATen/native/mkldnn/xpu/detail/Attr.h>
+#include <ATen/native/mkldnn/xpu/detail/Utils.h>
+
+#include <oneapi/dnnl/dnnl.hpp>
+#include <cstdint>
+
+
+namespace at::native::onednn {
+
+inline Tensor resize_as_onednn_mat1(const Tensor& mat1, const Tensor& output) {
+  auto output_ = output.flatten(0, -2);
+  int n = output_.sizes()[1];
+  auto sizes = mat1.sym_sizes().vec();
+  sizes[sizes.size() - 1] = n;
+  return output.view_symint(sizes);
+}
+
+sycl::event woq_matmul_int4(
```

Comment:
Suggest to add more function description here. e.g. the activation data type supported, data layout information for both inputs. etc...